python/packages/autogen-ext/src/autogen_ext/agents/semantic_kernel/_sk_assistant_agent.py

```diff
@@ -0,0 +1,327 @@
+
+        # 4) Invoke SK to get an assistant response
+        sk_responses = await chat_completion_service.get_chat_message_contents(
+            chat_history=self._chat_history,
+            settings=settings,
+            kernel=self._kernel,
+        )
+        # Convert SK's list of responses into a single final text
+        assistant_reply = "\n".join(r.content for r in sk_responses if r.content)
+        reply_message = TextMessage(content=assistant_reply, source=self.name)
+
+        # 5) Add the new assistant message into our chat history
+        if assistant_reply.strip():
+            self._chat_history.add_message(
+                self._convert_chat_message_to_sk_chat_message_content(AuthorRole.ASSISTANT, reply_message)
+            )
+
+        # 6) Return an autogen Response containing the text
+        return Response(chat_message=reply_message)
+
+    async def on_messages_stream(
+        self,
+        messages: Sequence[ChatMessage],
+        cancellation_token: CancellationToken,
+    ) -> AsyncGenerator[ChatMessage | Response, None]:
+        """
+        Handle new messages in streaming mode, yielding partial text messages
+        as we receive them, then yield a final single Response.
+        """
+        # 1) Convert & store new agent messages
+        for msg in messages:
+            sk_msg = self._convert_chat_message_to_sk_chat_message_content(AuthorRole.USER, msg)
+            self._chat_history.add_message(sk_msg)
+
+        # 2) Retrieve chat completion service
+        chat_completion_service = self._kernel.get_service(
+            service_id=self._service_id,
+            type=ChatCompletionClientBase,
+        )
+        if not chat_completion_service:
+            raise KernelServiceNotFoundError(f"Chat completion service not found with service_id: {self._service_id}")
+
+        assert isinstance(chat_completion_service, ChatCompletionClientBase)
+
+        settings = (
+            self._execution_settings
+            or self._kernel.get_prompt_execution_settings_from_service_id(self._service_id)
+            or chat_completion_service.instantiate_prompt_execution_settings(  # type: ignore
+                service_id=self._service_id,
+                extension_data={"ai_model_id": chat_completion_service.ai_model_id},
+            )
+        )
+
+        # 3) Stream the SK response
+        accumulated_reply: list[str] = []
+        async for sk_message_list in chat_completion_service.get_streaming_chat_message_contents(
+            chat_history=self._chat_history,
+            settings=settings,
+            kernel=self._kernel,
+        ):
+            for sk_message in sk_message_list:
+                # If it's streaming text, yield partial text as a new TextMessage
+                if sk_message.content:
+                    partial_text = sk_message.content
+                    accumulated_reply.append(partial_text)
+                    yield TextMessage(content=partial_text, source=self.name)
```

Comment:
To keep the behavior consistent with AssistantAgent, let's accumulate and output only complete messages rather than partial messages.